### ISSUE TITLE
Same performance without pretrained vision encoder

### ISSUE THREAD

**Issue by jiangranlv** [iMSquared/corn#4]
Hi, thank you for providing this amazing codebase and pretrained checkpoints! I am trying to reproduce the results of the paper, but I encountered an issue with the pretrained vision encoder checkpoint.

Specifically, I observed that when using the provided pretrained encoder checkpoint, the performance is almost identical to end-to-end rl training without the checkpoint. Both them performs well.

I wonder if I might have missed something in the setup. Could you please provide any guidance on what might be going wrong?

![Image](https://github.com/user-attachments/assets/9e5bcf9e-80f1-4713-b620-0caa8208f0d1)

**Comment by yycho0108** [comment]
@jiangranlv Can you share your training configuration (command line arguments, configuration file, etc.)? It's hard to tell what's going on just from the plot.

**Comment by jiangranlv** [comment]
![Image](https://github.com/user-attachments/assets/3082dd34-72cb-4d89-babe-01cfa12913ff)
The only modification to the original codebase is the addition of an option to freeze the vision encoder (/home/user/corn/pkm/scripts/train/envs/cube_env_wrappers.py:ICPEmbObs). The two experiments compare using the CORN checkpoint (freeze)versus not using it (no freeze). The other command lines arguments are identical to those in the README without any alterations.

**Comment by yycho0108** [comment]
Thank you for sharing your code. Actually, gradient doesn't flow to an ObservationWrapper, since it's considered part of the environment and not part of the policy. So disabling `cfg.frozen` in your case wouldn't actually train the encoder! It would only allow updates to its normalization statistics (due to `nn.Module.train()`, which would have minimal impact. This is why the training progression plot looks identical. In order to do end-to-end training, you'd have to move the encoder as part of the policy network (see, for instance, [point_patch_v5.py](pkm/src/pkm/models/rl/net/point_patch_v5.py))

**Comment by jiangranlv** [comment]
> Thank you for sharing your code. Actually, gradient doesn't flow to an ObservationWrapper, since it's considered part of the environment and not part of the policy. So disabling `cfg.frozen` in your case wouldn't actually train the encoder! It would only allow updates to its normalization statistics (due to `nn.Module.train()`, which would have minimal impact. This is why the training progression plot looks identical. In order to do end-to-end training, you'd have to move the encoder as part of the policy network (see, for instance, [point_patch_v5.py](pkm/src/pkm/models/rl/net/point_patch_v5.py))

Thanks a lot for your patient reply! We will modify our code carefully.
